File: scripts/free_action_runner.ts
#!/usr/bin/env bun

import { run } from "../run";

if (process.platform === "darwin") {
    process.exit(0);
}

run("sudo df -h");

const paths = [
    "/usr/share/dotnet",
    "/opt/ghc",
    "/usr/local/share/boost",
    process.env.AGENT_TOOLSDIRECTORY,
    "/usr/local/lib/android",
    "/opt/hostedtoolcache",
    "/__t/CodeQL",
];

for (const path of paths) {
    if (path) {
        run(`sudo rm -rf "${path}"`, false);
    }
}

run("sudo df -h");

File: scripts/free_action_runner.rs
#!/usr/bin/env rust

use anyhow::Result;
use shared::run::{run, run_allow_fail};

fn main() -> Result<()> {
    if cfg!(target_os = "macos") {
        return Ok(());
    }

    run("sudo df -h")?;

    let mut paths = vec![
        "/usr/share/dotnet".to_string(),
        "/opt/ghc".to_string(),
        "/usr/local/share/boost".to_string(),
        "/usr/local/lib/android".to_string(),
        "/opt/hostedtoolcache".to_string(),
        "/__t/CodeQL".to_string(),
    ];
    if let Ok(tools) = std::env::var("AGENT_TOOLSDIRECTORY") {
        paths.push(tools);
    }

    for path in &paths {
        run_allow_fail(&format!("sudo rm -rf \"{path}\""));
    }

    run("sudo df -h")?;
    Ok(())
}
